### PR TITLE
Explore: Changed references to DataQuery and DataSourceRef

### LINKED ISSUE #61946: Change references to DataQuery and DataSourceRef
As of #61192, the exported types from `packages/grafana-data/src/types/query.ts` were deprecated in favor of using types from grafana/schema ([src](https://github.com/grafana/grafana/pull/61192/files#diff-2c9e457cb87628517f9e0606e5d774027ff002171070b0d561de1a178bf26e15))

In the `public/app/features/explore` directory, where ever `DataQuery` and `DataSourceRef` are imported from `@grafana/data` - they should be changed to be imported from `@grafana/schema`

For pointing, we use these types extensively in Explore (79 times in 19 files for DataQuery and 9 results in 4 files for DataSourceRef)

### PR DESCRIPTION
**Pull request by aashish0909**
**What is this feature?**

This PR changes the references to DataQuery and DataSourceRef in the `public/app/features/explore` directory from `@grafana/data` to `@grafana/schema`.

**Why do we need this feature?**

As of #61192, the exported types from `packages/grafana-data/src/types/query.ts` were deprecated in favor of using types from grafana/schema ([src](https://github.com/grafana/grafana/pull/61192/files#diff-2c9e457cb87628517f9e0606e5d774027ff002171070b0d561de1a178bf26e15)). So we are changing the references.

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/61946

### CODE REVIEW

**Comment by aashish0909** [comment]
What should I do to fix this Enterprise PR error @gelicia ?

**Review by gelicia — APPROVED** [review]
Checked out fork, all imports seem to be accounted for and in the correct order for the linter. Thanks for your contribution!

**Comment by aashish0909** [comment]
> Checked out fork, all imports seem to be accounted for and in the correct order for the linter. Thanks for your contribution! 

But would this get merged? It's showing 2 errors

**Comment by gelicia** [comment]
> What should I do to fix this Enterprise PR check error @gelicia ?
> But would this get merged? It's showing 2 errors

It's still building.

The enterprise error message says `Token is empty, can't dispatch event. This is expected for PRs coming from forks, please check that the changes are compatible with Enterprise before merging this PR.` - I verify changing these imports should be compatible with Enterprise, so I will be sure this doesn't hold up this PR. Thank you for checking! 

TL;DR: I'll take care of it, no worries.

**Comment by aashish0909** [comment]
Thank you! 
Can you suggest me something to do next?
I am new to open source contribution.

**Comment by gelicia** [comment]
> Thank you! Can you suggest me something to do next? I am new to open source contribution.

I believe whatever tag you were looking at to find this ticket is the right way to go, unfortunately I can't think of anything I've added recently. Good luck and thanks again for your help!

**Pull request merged**
